marklogic-client-api/src/main/java/com/marklogic/client/dataservices/InputEndpoint.java

```diff
@@ -0,0 +1,15 @@
+package com.marklogic.client.dataservices;
+
+import com.marklogic.client.DatabaseClient;
+import com.marklogic.client.dataservices.impl.InputEndpointImpl;
+import com.marklogic.client.io.marker.JSONWriteHandle;
+
+public interface InputEndpoint {
+	
+	public interface BulkInputCaller{
+		
+	}
+	static InputEndpoint on(DatabaseClient client, JSONWriteHandle apiDecl) {
```

Comment:
It might be a good convention to keep the static factory at the top of the interface so it is easy to find